client/interfaces.go

```diff
@@ -82,6 +88,20 @@ func (client *Client) Interfaces() (interfaces Interfaces, err error) {
 	return
 }
 
+// InterfaceIndex returns the list of all interface names.
+//
+// The result will only have three fields populated, Name, Summary and Used.
+func (client *Client) InterfaceIndex() (ifaces []Interface, err error) {
```

Comment:
This makes total sense, on 2nd read, with fresh head. I'm mid-way through the implementation. Hopefully will be in a reviewable state by EOD today.